### PR TITLE
Implicit m2 m constraint name length fix

### PR DESCRIPTION
**Pull request by do4gr**
Fixes https://github.com/prisma/prisma/issues/10094

### CODE REVIEW

File: libs/datamodel/core/src/configuration/configuration_struct.rs [hunk 1]
```diff
@@ -29,6 +29,13 @@ impl Configuration {
         self.datasources.first().map(|source| source.referential_integrity())
     }
 
+    pub fn max_identifier_length(&self) -> usize {
+        self.datasources
+            .first()
+            .map(|source| source.active_connector.max_identifier_length())
+            .unwrap_or(usize::MAX)
+    }
```
Comment (tomhoule) [inline libs/datamodel/core/src/configuration/configuration_struct.rs:37]: For this one, if I remember the API correctly, there is `configuration.active_connector()` or `datasource.active_connector()`, which is always defined and defaults to `EmptyConnector`. I don't know if the worry is founded, but `configuration.active_connector().max_identifier_length()` and `configuration.max_identifier_length()` could then be different.

**Review by garrensmith — APPROVED** [review]

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_calculator.rs [hunk 2]
```diff
@@ -210,7 +223,10 @@ fn calculate_relation_tables<'a>(
             ];
 
             sql::Table {
-                name: table_name,
+                name: table_name
+                    .chars()
+                    .take(configuration.max_identifier_length() - 1)
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_calculator.rs:228]: I don't understand the `-1`, intuitively taking the max identifier length should result in the length being the maximum, but still ok, right?

**Review by tomhoule — COMMENTED** [review]
My only worry is that we find a connector with different maximum lengths for different kinds of identifiers, but we can deal with it when we encounter it.

I'm not approving right away only because of the question around the `-1`s in sql-schema-calculator.

**Review by tomhoule — APPROVED** [review]

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_calculator.rs (hunk 2 above)
Comment (do4gr) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_calculator.rs:228]: Good spot, that is wrong.

**Pull request merged**